### PR TITLE
2.5.1 login service

### PR DESCRIPTION
**Pull request by emadvaid**

### CODE REVIEW

File: gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/config/AdminUserPreload.java [hunk 1]
```diff
@@ -49,13 +49,13 @@ public void run(ApplicationArguments args) {
             }
 
             // first check if the admin user is in the database
-            if(!userDao.findByEmail("admin").isPresent()) {
+            if(!userDao.findByEmail("admin@aaa.com").isPresent()) {
```
Comment (sloanlipman) [inline gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/config/AdminUserPreload.java:52]: Does this need to be hardcoded?

File: gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/config/AdminUserPreload.java [hunk 2]
```diff
@@ -36,7 +36,7 @@ public void run(ApplicationArguments args) {
         if(createAdmin!=null && "true".equals(createAdmin.trim().toLowerCase())) {
 
             // defaults for the "bootstrapping admin"
-            String email = "admin@admin.com";
+            String email = "admin";
             String password = "password";
```
Comment (sloanlipman) [inline gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/config/AdminUserPreload.java:40]: remove hardcoded values?

File: gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/config/AdminUserPreload.java [hunk 3]
```diff
@@ -49,13 +49,13 @@ public void run(ApplicationArguments args) {
...
             // first check if the admin user is in the database
-            if(!userDao.findByEmail("admin").isPresent()) {
+            if(!userDao.findByEmail("admin@aaa.com").isPresent()) {
                 // create location for admin user
                 Location adminLocation = new Location();
                 adminLocation = locDao.save(adminLocation);
                 // create the admin user
                 UserRec adminUserRec = new UserRec(
-                        "admin@aaa.com", "admin", null,"password",
+                        "admin@aaa.com", "admin", "admin","password",
```
Comment (sloanlipman) [inline gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/config/AdminUserPreload.java:58]: more hardcoding - is it needed?

File: gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/model/Location.java [hunk 4]
```diff
@@ -41,13 +46,14 @@ public Location(Location original) {
         this.lat = original.lat;
         this.lng = original.lng;
         this.locationString = original.locationString;
+        this.userRecSet = new HashSet<>();
     }
 
-    public UUID getId() {
+    public int getId() {
```
Comment (sloanlipman) [inline gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/model/Location.java:52]: this same method is in UserDetail.java. Any way you can encapsulate some of these methods without causing too much trouble?

File: matchmaker-ui/package.json [hunk 5]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.1",
   "scripts": {
     "ng": "ng",
-    "start": "ng serve",
+    "start": "ng serve --proxy-config proxy.conf.json",
```
Comment (sloanlipman) [inline matchmaker-ui/package.json]: Add --open to the "start" script and see if it automatically opens localhost:4200. If that works, add that change.

File: matchmaker-ui/src/app/pages/login-page/login-page.component.ts [hunk 6]
```diff
@@ -13,22 +13,27 @@ import { LoginService } from 'src/app/shared/services/login-service/login.servic
...
-  loginService: LoginService;
+export class LoginPageComponent implements OnInit {
+
   constructor(
     protected router: Router,
     protected location: Location,
     protected injector: Injector,
-    protected dialog: MatDialog
+    protected dialog: MatDialog,
+    protected   loginService: LoginService
```
Comment (sloanlipman) [inline matchmaker-ui/src/app/pages/login-page/login-page.component.ts]: fix spacing between protected and loginService

File: matchmaker-ui/src/app/pages/login-page/login-page.component.ts [hunk 7]
```diff
@@ -13,22 +13,27 @@ import { LoginService } from 'src/app/shared/services/login-service/login.servic
   encapsulation: ViewEncapsulation.None
 
 })
-export class LoginPageComponent extends AppComponent implements OnInit {
```
Comment (sloanlipman) [inline matchmaker-ui/src/app/pages/login-page/login-page.component.ts:15]: why did you remove the inheritance?

File: matchmaker-ui/src/app/shared/services/login-service/login.service.ts [hunk 8]
```diff
@@ -20,18 +20,22 @@ export class LoginService /* extends HttpService */ { // TODO decide if there ne
...
     protected http: HttpClient
     ) {
       // super(http);
     }
 
+
+  private httpOptions = {
+    headers: new HttpHeaders({
+      'Content-Type':  'application/json',
+      'Access-Contro': 'Access-Control-Allow-Headers'
```
Comment (sloanlipman) [inline matchmaker-ui/src/app/shared/services/login-service/login.service.ts]: Looks like you lost the last letter of Control

**Review by sloanlipman — COMMENTED** [review]
Sorry, I forgot to hit submit. Here are the comments.

File: matchmaker-ui/src/app/pages/login-page/login-page.component.ts (hunk 7 above)
Comment (emadvaid) [reply matchmaker-ui/src/app/pages/login-page/login-page.component.ts:15]: it was using the proxy config for the frontend code as well its going from localhost:4200 /login to localhost:8080/login.  Once i removed it that error went away.

File: gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/config/AdminUserPreload.java (hunk 2 above)
Comment (emadvaid) [reply gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/config/AdminUserPreload.java:40]: No this is called admin preload, it automatically makes an admin from the backend.  Admin is  the first one to login the system and create more users.

File: matchmaker-ui/src/app/pages/login-page/login-page.component.ts (hunk 7 above)
Comment (sloanlipman) [reply matchmaker-ui/src/app/pages/login-page/login-page.component.ts:15]: Ah ok. That's because I have some duplicate naming stuff that I'm working on. I think this is going to be fine as-is, and I can fix it in future builds.

File: .gitignore [hunk 9]
```diff
@@ -1,3 +1,4 @@
 node_modules
 .DS_Store
 Thumbs.db
+/localCopyMatchmaker
```
Comment (sloanlipman) [inline .gitignore:4]: Remove this. I know I added it before, but we don't actually need it.

Comment (emadvaid) [reply .gitignore:4]: gitignore

Comment (emadvaid) [reply .gitignore:4]: i don't see anything

File: matchmaker-ui/src/app/shared/services/login-service/login.service.ts (hunk 8 above)
Comment (emadvaid) [reply matchmaker-ui/src/app/shared/services/login-service/login.service.ts]: fixed

File: matchmaker-ui/package.json (hunk 5 above)
Comment (emadvaid) [reply matchmaker-ui/package.json]: it worked, i added it

File: gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/model/Location.java (hunk 4 above)
Comment (emadvaid) [reply gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/model/Location.java:52]: Id is a variable of the class, I will look into it.

File: matchmaker-ui/package.json (hunk 5 above)
Comment (sloanlipman) [reply matchmaker-ui/package.json]: awesome

File: gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/model/Location.java (hunk 4 above)
Comment (sloanlipman) [reply gamingMatchMaker/src/main/java/com/gamingMatchMaker/gamingMatchMaker/model/Location.java:52]: Yeah actually just leave that one alone. I wasn't really paying attention to that when I wrote this comment. It's just a getter method, so it shouldn't be encapsulated.

**Review by sloanlipman — APPROVED** [review]
